### ISSUE TITLE
`allowArbitraryExtensions` does not work as expected

### ISSUE THREAD

**Issue by indooorsman** [microsoft/TypeScript#55507]
### 🔎 Search Terms

allowArbitraryExtensions

### 🕗 Version & Regression Information

- This is unexpected error from Typescript@5.0

### ⏯ Playground Link

https://codesandbox.io/p/sandbox/typescript-playground-export-forked-mfz7rg?file=/index.ts:10,28

### 💻 Code

```ts
import {} from "./app.css";

export {};
```
my tsconfig.json
```json
{
  "compileOnSave": false,
  "compilerOptions": {
    "module": "ESNext",
    "target": "ESNext",
    "lib": [
      "ESNext",
      "DOM",
      "DOM.Iterable"
    ],
    "moduleResolution": "bundler",
    "customConditions": [
      "dev"
    ],
    "jsx": "react-jsx",
    "declaration": true,
    "emitDeclarationOnly": true,
    "isolatedModules": true,
    "strict": true,
    "allowJs": true,
    "skipDefaultLibCheck": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "allowArbitraryExtensions": true,
    "allowImportingTsExtensions": true,
    "resolveJsonModule": true,
    "sourceMap": false,
    "rootDirs": [
      "./src",
      "./esm"
    ],
    "outDir": "./esm"
  },
  "include": [
    "./src/**/*"
  ]
}

```

### 🙁 Actual behavior

I have set `allowArbitraryExtensions` to true, but it still throw error when import a `.css` file.
As mentioned in document:
>  you can **_suppress_** the error with the new `--allowArbitraryExtensions` compiler option.

### 🙂 Expected behavior

should not throw error

### Additional information about the issue

![issue](https://github.com/microsoft/TypeScript/assets/4919405/c2068d9b-13f6-4486-a633-27c219b8d9bf)

**Comment by MartinJohns** [comment]
You're missing the `app.d.css.ts` file.

**Comment by indooorsman** [comment]
doesn't it mean any errors about this file with unknown extension would be suppressed ?
________________________________
发件人: Martin Johns ***@***.***>
发送时间: Friday, August 25, 2023 4:23:17 PM
收件人: microsoft/TypeScript ***@***.***>
抄送: indooorsman ***@***.***>; Author ***@***.***>
主题: Re: [microsoft/TypeScript] `allowArbitraryExtensions` does not work as expected (Issue #55507)


You're missing the app.d.css.ts file.

―

**Comment by fatcerberus** [comment]
Not sure where you got that idea.

https://github.com/microsoft/TypeScript/pull/51435, where this was implemented:

> Add `--allowArbitraryExtensions`, a flag for allowing arbitrary extensions on import paths

> And, correspondingly, **the `.d.ext.ts` files needed** to describe the shape of whatever it is [you’re] importing.

You still need types for the file you’re importing, as with any other import.  The flag just prevents the import from failing on the basis of the file extension alone.

**Comment by indooorsman** [comment]
Got it, thanks for the explanation
________________________________
发件人: Bruce Pascoe ***@***.***>
发送时间: Friday, August 25, 2023 9:07:30 PM
收件人: microsoft/TypeScript ***@***.***>
抄送: indooorsman ***@***.***>; Author ***@***.***>
主题: Re: [microsoft/TypeScript] `allowArbitraryExtensions` does not work as expected (Issue #55507)


Not sure where you got that idea.

#51435<https://github.com/microsoft/TypeScript/pull/51435>, where this was implemented:

Add --allowArbitraryExtensions, a flag for allowing arbitrary extensions on import paths

And, correspondingly, the .d.ext.ts files needed to describe the shape of whatever it is [you’re] importing.

You still need types for the file you’re importing, as with any other import. The flag just prevents the import from failing on the basis of the file extension alone.

―
